src/main/scala/se/nimsa/dcm4che/streams/DicomFlows.scala

```diff
@@ -172,24 +172,26 @@ object DicomFlows {
 
       case _: DicomItem if discarding => Nil
       case _: DicomItemDelimitation if discarding => Nil
-      case _: DicomFragmentsDelimitation if discarding =>
-        discarding = false
-        Nil
+      case _: DicomFragmentsDelimitation if discarding => Nil
+
+      case _: DicomSequence if discarding => Nil
+      case _: DicomSequenceDelimitation  if discarding => Nil
 
       case dicomAttribute: DicomAttribute  =>
-        discarding = shouldDiscard(dicomAttribute.header.tag, dicomAttribute.header.isFmi, applyToFmi, isWhitelist)
+        discarding = shouldDiscard(dicomAttribute.header.tag, isWhitelist)
         if (discarding) {
           Nil
         } else {
           dicomAttribute :: Nil
         }
 
-      case dicomPart =>
+      case dicomPart => // FIXME: Unknown part?
```

Comment:
missed that one - would it be better to discard for a whitelist and leave it for a blacklist filter?